addons/hr_holidays/report/hr_leave_employee_report.py

```diff
@@ -0,0 +1,176 @@
+    working_schedule_aligned_date_to = fields.Datetime('Date To', compute='_compute_working_schedule_aligned_dates', readonly=True, store=True)
+    number_of_days = fields.Float(compute='_compute_leave_duration', readonly=True, store=True)
+    number_of_hours = fields.Float(compute='_compute_leave_duration', readonly=True, store=True)
+    description = fields.Char()
+    holiday_status_id = fields.Many2one("hr.leave.type", string="Time Off Type")
+    state = fields.Selection([
+        ('confirm', 'To Approve'),
+        ('refuse', 'Refused'),
+        ('validate1', 'Second Approval'),
+        ('validate', 'Approved'),
+        ('cancel', 'Cancelled'),
+    ])
+    color = fields.Integer(string="Color", related='holiday_status_id.color')
+
+    @property
+    def _table_query(self):
+        fetched_leave_field_names, leave_records = self._fetch_leave_data()
+        report_records = self._create_report_records_from_leave_records(leave_records, fetched_leave_field_names)
+        self._compute_working_schedule_aligned_dates(report_records)
+        self._compute_leave_duration(report_records)
+        return self._generate_report_query(report_records)
+
+    def _fetch_leave_data(self):
+        self.env.cr.execute(SQL(
+            """
+            WITH leave_data AS (
+                SELECT
+                    ROW_NUMBER() OVER(ORDER BY employee_id, days_included_in_request) AS id,
+                    id AS leave_id, employee_id, date_from, date_to,
+                    holiday_status_id, state, private_name AS description,
+                    DATE_TRUNC('day', days_included_in_request) AS day_start
+                FROM hr_leave hl
+                CROSS JOIN LATERAL GENERATE_SERIES(
+                    DATE_TRUNC('day', date_from),
+                    DATE_TRUNC('day', date_to),
+                    INTERVAL '1 day'
+                ) AS days_included_in_request
+                WHERE hl.employee_company_id IN %(company_ids)s
+            )
+            SELECT
+                id, leave_id, employee_id,
+                holiday_status_id, state, description,
+                GREATEST(date_from, day_start) AS day_aligned_date_from,
+                LEAST(date_to, (day_start + INTERVAL '1 day' - INTERVAL '1 second')) AS day_aligned_date_to
+            FROM leave_data;
+            """,
+            company_ids=tuple(self.env.companies.ids),
+        ))
+        fetched_leave_field_names = [desc[0] for desc in self.env.cr.description]
+        leave_records = self.env.cr.fetchall()
+        return fetched_leave_field_names, leave_records
+
+    def _create_report_records_from_leave_records(self, leave_records, leave_field_names):
+        report_records = []
+        for leave_record in leave_records:
+            report_records.append({field_name: leave_record[index] for index, field_name in enumerate(leave_field_names)})
+
+        return report_records
+
+    def _compute_working_schedule_aligned_dates(self, report_records):
```

Comment:
Do we need this? I don't feel like this is giving added value and it relies on some currently heavy computation